### DISCUSSION TITLE
Parsing errors detected when parsing ${filterParameters}

### DISCUSSION THREAD

**Discussion by UBSusmitha** [SAP/open-ux-odata#690] [Q&A]
After we started using fe mockserver for our local apps in ovp, while filtering some of the values from smart filter bar we get this error in the terminal which is thrown by ovp/node_modules/@sap-ux/fe-mockserver-core/dist/request/filterParser.js

<img width="799" alt="MicrosoftTeams-image" src="https://github.com/SAP/open-ux-odata/assets/140173458/9043ab0f-9fb9-4ced-950b-3f90ab2070d4">

Filter gets applied correctly & cards load when we remove this if block  if (parser.error.length) {} & restart the server.

**Steps to reproduce:**
Execute yarn start in the ovp directory.
By default app will run on localhost:8080 unless the port is occupied. If port is occupied the application will run on next available port.
After applying filter to NetAmount = 16474, it shows internal server error with the Parsing error detected when parsing error in the terminal, it works fine if this if block is removed & server is restarted.
<img width="1792" alt="Screenshot 2023-09-11 at 4 06 51 PM" src="https://github.com/SAP/open-ux-odata/assets/140173458/b495f0b8-45b0-4e67-885e-84c43aad7374">

Could you please let us know if there is any way to resolve this instead of this work around?

**Comment by nlunets** [comment]
Can you provide the queries that are failing so that we can analyze them ?

**Comment by UBSusmitha** [comment]
This is one of them: "SalesOrderLineItemSet/$count?$filter=(DeliveryDate ge datetime'2023-04-01T00:00:00' and DeliveryDate le datetime'2023-06-30T00:00:00') and NetAmount eq 16474m"

**Comment by nlunets** [comment]
Hmmm, why do you have NetAmount eq 16474m what's that `m` for ?

**Reply by UBSusmitha** [reply]
In ovp we are sending correct values of filter received from SFB which is 16474 & is not appended with m, its getting added from:
[http://localhost:8080/resources/sap/ui/model/odata/ODataUtils.js?eval](url)

OVP > Main.controller.js > _overrideCardModelRead() > _that._addRelevantFilters(); >
var sFilterParams = ODataUtils.createFilterParams().substr(8); > sap/ui/model/odata/ODataUtils.js > this._createFilterParams() > create() > that._createFilterSegment() > this.formatValue() > Switch case "Edm.Decimal": sValue = vValue + "m";

But It works if this is removed from here & server is restarted again:
ovp/node_modules/@sap-ux/fe-mockserver-core/dist/request/filterParser.js

```
if (parser.errors.length > 0) {
    parser.errors.forEach(console.error);
    throw new Error(`Parsing errors detected when parsing ${filterParameters}`)
}

```

**Reply by nlunets** [reply]
Ok that's part of the v2 protocol then, i need to look into it.

Removing the part of the code that checks for error while parsing is not a good idea ;)
